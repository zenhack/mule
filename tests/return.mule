#XFAIL: type formatting
fn x. x {}
