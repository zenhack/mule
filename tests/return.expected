
Desugared:

	(fn x (x {}))

inferred type:

	(all a (((record (types ()) (values ())) -> a) -> a))

Runtime term:

	(Lam (0 () (App ((Var 0) (Record ())))))

Evaluated:

	(Lam (0 () (App ((Var 0) (Record ())))))
